src/components/TrackingConsent.vue

```diff
@@ -0,0 +1,467 @@
+    })
+    public geoIpServer: string;
+
+    private uiRequired: boolean = false;
+
+    public static get _paq() {
+        if (!window._paq || !Array.isArray(window._paq)) {
+            window._paq = [];
+        }
+        return window._paq;
+    }
+
+    public static get _mtm() {
+        if (!window._mtm || !Array.isArray(window._mtm)) {
+            window._mtm = [];
+        }
+        return window._mtm;
+    }
+
+    /**
+     * consent getter - return the parsed content of the consent cookie that store the user's choice about data sharing
+     * @return {Consent} An object containing two boolean properties: allowsBrowserData & allowsUsageData
+     */
+    public static get consent(): Consent {
+        const cookie = TrackingConsent._getCookie(TrackingConsent.COOKIE_STORAGE_KEY);
+
+        if (cookie) {
+            return JSON.parse(cookie);
+        }
+
+        return {};
+    }
+
+    public static get allowsUsageData(): boolean {
+        return TrackingConsent.consent.allowsUsageData;
+    }
+
+    public static get allowsBrowserData(): boolean {
+        return TrackingConsent.consent.allowsBrowserData;
+    }
+
+    /**
+     * trackEvent - allow you to track custom interaction on the website/webapp.
+     *
+     * This method won't do anything if the user denied to share usage data.
+     *
+     * Docs: https://matomo.org/docs/event-tracking/
+     */
+    public static trackEvent(
+        category: string,
+        action: string,
+        name?: string,
+        value?: string | number,
+    ) {
+        if (!TrackingConsent.allowsUsageData) return;
+
+        const obj: Array<string | number> = ['trackEvent', category, action];
+
+        if (name) {
+            obj.push(name);
+        }
+        if (value) {
+            obj.push(value);
+        }
+
+        TrackingConsent._paq.push(obj);
+    }
+
+    /**
+     * execFunction - allow you to execute a function in matomo's scope, where `this` is Matomo's tracker object
+     *
+     * Docs: https://developer.matomo.org/guides/tracking-javascript-guide
+     * API ref: https://developer.matomo.org/guides/tracking-javascript
+     */
+    public static execFunction(fn: () => void) {
+        TrackingConsent._paq.push([fn]);
+    }
+
+    private static _setCookie(
+        cookieName: string,
+        cookieValue: string,
+        options: {
+            domain: string,
+            expirationDays: number,
+        },
+    ) {
+        const cookie = [cookieName + '=' + cookieValue];
+
+        if (options) {
+            if (!options.domain) options.domain = TrackingConsent.DEFAULT_COOKIE_DOMAIN;
+            if (!options.expirationDays) options.expirationDays = TrackingConsent.DEFAULT_COOKIE_EXPIRATION_DAYS;
+
+            cookie.push('domain=' + options.domain);
+            cookie.push('max-age=' + options.expirationDays * 24 * 60 * 60);
+        }
+
+        cookie.push('path=/');
+        document.cookie = cookie.join(';');
```

Comment:
I believe this needs the `sameSite` setting as well nowadays.

Reply:
`SameSite` require `Secure` sometimes, so i'll add both 👍